### ISSUE TITLE
[<Library component: Model|Core|etc...>] Compatibility issue with Pytorch 2.7.0

### ISSUE THREAD

**Issue by nogader** [Nixtla/neuralforecast#1336]
### What happened + What you expected to happen

The following new fix in version 3.0.1:
FIX: Set 2.0.0<=pytorch<=2.6.0 to avoid conflicts with networkx with Python 3.9 @marcopeix (https://github.com/Nixtla/neuralforecast/pull/1318)

Version 3.0.1 forces to use Pytorch version 2.6.0 or lower.  Is there a way to satisfy both issues?

Thanks.

### Versions / Dependencies

OS: Windows 11
Python: 3.11
Pytorch: 2.7.0
neuralforecast: 3.0.1 

### Reproduction script

Install Pytorch 2.7.0
pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu128

Then install neuralforecast 3.0.1 
pip3 install neuralforecast

### Issue Severity

None

**Comment by marcopeix** [comment]
If you are using Python > 3.9, then you should be able to work with PyTorch 2.7. This constraint was implemented for Python 3.9, as we still support that version for now.

**Comment by nogader** [comment]
It does not work on python 3.11 with PyTorch 2.7.0.  

Let me show the steps to reproduce the error;

1. conda create --name test python=3.11 
2. conda activate test 
3. pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu128
4. pip install nixtlats neuralforecast datasetsforecast statsforecast mlforecast ray


During the last step, the pip attempts to uninstall PyTorch 2.7.0 and re-installs version 2.6.0.  Then the following error occurs;

**ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
torchaudio 2.7.0+cu128 requires torch==2.7.0+cu128, but you have torch 2.6.0 which is incompatible.
torchvision 0.22.0+cu128 requires torch==2.7.0+cu128, but you have torch 2.6.0 which is incompatible.**
Successfully installed Mako-1.3.10 adagio-0.2.6 aiohappyeyeballs-2.6.1 aiohttp-3.12.4 aiosignal-1.3.2 alembic-1.16.1 annotated-types-0.7.0 anyio-4.9.0 appdirs-1.4.4 attrs-25.3.0 certifi-2025.4.26 charset-normalizer-3.4.2 click-8.2.1 cloudpickle-3.1.1 colorama-0.4.6 colorlog-6.9.0 coreforecast-0.0.16 datasetsforecast-1.0.0 frozenlist-1.6.0 fs-2.4.16 fugue-0.9.1 greenlet-3.2.2 h11-0.16.0 httpcore-1.0.9 httpx-0.28.1 idna-3.10 joblib-1.5.1 jsonschema-4.24.0 jsonschema-specifications-2025.4.1 lightning-utilities-0.14.3 llvmlite-0.44.0 mlforecast-1.0.2 msgpack-1.1.0 multidict-6.4.4 neuralforecast-3.0.1 nixtlats-0.5.2 numba-0.61.2 optuna-4.3.0 packaging-25.0 pandas-2.2.3 patsy-1.0.1 propcache-0.3.1 protobuf-6.31.1 pyarrow-20.0.0 pydantic-2.11.5 pydantic-core-2.33.2 python-dateutil-2.9.0.post0 pytorch-lightning-2.5.1.post0 pytz-2025.2 pyyaml-6.0.2 ray-2.46.0 referencing-0.36.2 requests-2.32.3 rpds-py-0.25.1 scikit-learn-1.6.1 scipy-1.15.3 six-1.17.0 sniffio-1.3.1 sqlalchemy-2.0.41 statsforecast-2.0.1 statsmodels-0.14.4 sympy-1.13.1 tenacity-9.1.2 tensorboardX-2.6.2.2 threadpoolctl-3.6.0 torch-2.6.0 torchmetrics-1.7.2 tqdm-4.67.1 triad-0.9.8 typing-inspection-0.4.1 tzdata-2025.2 urllib3-2.4.0 utilsforecast-0.2.12 xlrd-2.0.1 yarl-1.20.0

**Comment by marcopeix** [comment]
Did you try installing a newer version of PyTorch after installing neuralforecast?

**Comment by marcopeix** [comment]
Then, you have to stick to PyTorch 2.6, because that's a requirement we implemented to support Python 3.9.
